src/batoid.cpp

```diff
@@ -67,6 +68,36 @@ namespace batoid{
         return RayVector(std::move(result), wavelength);
     }
 
+    RayVector uniformCircularGrid(double dist, double outer, double inner,
+                           double xcos, double ycos, double zcos,
+                           int nrays, double wavelength, double flux, const Medium& m, int seed) {
+        double n = m.getN(wavelength);
+
+        std::vector<Ray> result;
+        result.reserve(nrays);
+
+        // The "velocities" of all the rays in the grid are the same.
+        Vector3d v(xcos, ycos, zcos);
+        v.normalize();
+        v /= n;
+
+        // Instantiate uniform distribution.
+        std::default_random_engine generator(seed);
+        std::uniform_real_distribution<double> distribution(0.0, 1.0);
+
+        for (int i=0; i<nrays; i++) {
+            // Draw radius, azimuth angle uniformly over annulus.
+            double mu = distribution(generator);
+            double nu = distribution(generator);
+            double radius = sqrt(mu * pow(outer, 2) + (1 - mu) * pow(inner, 2));
```

Comment:
The compiler probably optimizes this the same anyway, but just in case not, I think this could be made more efficient using
`pow(outer, 2)` -> `outer*outer` and similarly for inner.